### ISSUE TITLE
Selecting an item from a no-flow axis

### ISSUE THREAD

**Issue by henryiii** [scikit-hep/boost-histogram#274]
Currently, we do not have a single-item selection syntax in UHI. Something like this:

```python
value_ax = bh.axis.Regular(100, -5, 5)
bool_ax = bh.axis.Integer(0, 2, underflow=False, overflow=False)
run_number_ax = bh.axis.Category([2, 4, 7])

h = bh.Histogram(value_ax, bool_ax, run_number_ax)
```

```python
h[:, True, :]
h[:, 1, :]
h[:, 1:2:bh.sum, :] # Currently works
```

Not clear how to do this for the 3rd axis currently.

Also, a list selection syntax, for categories. So select 2 and 7, but not 4, for example. It would return a new, smaller category axis. Once we get full UHI, this could be implemented through a UHI object. Or we could come up with a new syntax, like `[2,7]::bh.sum` or something like that. Note that in Numpy, this would be a list, but there's no need to combine lists and slices, but we would often like to combine lists and actions.

**Comment by henryiii** [comment]
Edit: "Currently works" actually does not work, due to boostorg/histogram#257 . Once you turn off flow, histograms become unslicable.

**Comment by henryiii** [comment]
The first part of this is available in #279. The second part probably needs upstream support. This does not work on `*flow=False` axes, waiting for upstream fixes.

**Comment by henryiii** [comment]
Fixed upstream and in develop.
